docs/source/en/basics/router.md

```diff
@@ -0,0 +1,328 @@
+title: Router
+---
+
+Router is mainly used to describe the relationship between the request URL and the Controller that processes the request eventually. All routing rules are unified in the `app/router.js` file by the framework.
+
+By unifying routing rules, routing logics are free from scatter, random conflicts disappear and it becomes easy for us to see global routing rules.
+
+## How to Define Router
+
+- Define the routing rule in `app/router.js` file
+
+```js
+// app/router.js
+module.exports = app => {
+  app.get('/user/:id', 'user.info');
+};
+```
+
+- Implement the Controller in `app/controller` directory
+
+```js
+// app/controller/user.js
+exports.info = function* (ctx) {
+  ctx.body = {
+    name: `hello ${ctx.params.id}`,
+  };
+};
+```
+This simplest Router is done by now, when users do the request `GET /user/123`, the info function in `user.js` will be invoked.
+
+## Router config in detail
+
+Below is the complete definition of router, parameters can be determined depending on different scenes.
+
+```js
+app.verb('path-match', 'controller.action');
+app.verb('router-name', 'path-match', 'controller.action');
+app.verb('path-match', middleware1, ..., middlewareN, 'controller.action');
+app.verb('router-name', 'path-match', middleware1, ..., middlewareN, 'controller.action');
+```
+The complete definition of router includes 5 major parts:
+
+- verb - actions that users trigger, including get, post and so on, and will be explained in detail later.
+  * app.head - HEAD
+  * app.options - OPTIONS
+  * app.get - GET
+  * app.put - PUT
+  * app.post - POST
+  * app.patch - PATCH
+  * app.delete - DELETE
+  * app.del - this is a alias method due to the reservation of delete.
+  * app.redirect - redirect the request URL. For example, the most common case is to redirect the request accessing the root directory to the homepage.
```

Comment:
How about 'redirects the request URL' which means 'app.redirect redirects the request URL'?